lab_2_tokenize_by_bpe/main.py

```diff
@@ -58,7 +110,35 @@ def train(
     :param num_merges: required number of new tokens
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if not (isinstance(word_frequencies, dict) and
+            isinstance(num_merges, int)
+    ):
+        return None
+
+    count_of_pairs = count_tokens_pairs(word_frequencies)
+    if count_of_pairs is None:
+        return None
+    num_merges = min(num_merges, len(count_of_pairs))
+
+    for i in range(num_merges):
+        maximum = max(count_of_pairs.values())
+
+        local_maximums = [list(key) for key in count_of_pairs if count_of_pairs[key] == maximum]
+        max_length = max(len(''.join(tokens)) for tokens in local_maximums)
+
+        longest_pairs = [p for p in local_maximums if len(''.join(p)) == max_length]
+        pair = sorted(longest_pairs)[0]
```

Comment:
It looks like three previous lines could be expressed as a single `key` argument to `sorted`, could they?